torch/_inductor/runtime/hints.py

```diff
@@ -127,3 +127,33 @@ def create(cls, device):
             index=device.index,
             cc=device_interface.get_compute_capability(device),
         )
+
+
+class HalideInputSpec(typing.NamedTuple):
+    ctype: str
+    name: str
+    numel: Optional[str] = None
+
+    def bindings_type(self):
+        if self.ctype == "half*":
+            return "void*"  # half not defined
+        return self.ctype
+
+    def halide_type(self):
+        if self.ctype == "half*":
+            return "halide_type_t(halide_type_float, 16)"  # half not defined
+        return f"halide_type_of<{self.ctype.replace('*', '')}>()"
+
+
+class HalideMeta(typing.NamedTuple):
+    argtypes: List[HalideInputSpec]
+    target: str
+    scheduler: str
+    scheduler_flags: Dict[str, str]
```

Comment:
nit: I see the flags are all integer in the unit test. Maybe change the type annotation to:

```
Dict[str, Union[str, int]]
```